runtime/calamari/tests/xcm_tests.rs

```diff
@@ -0,0 +1,1791 @@
+mod xcm_mock;
+
+use codec::Encode;
+use frame_support::{assert_ok, weights::constants::WEIGHT_PER_SECOND};
+use manta_primitives::AssetLocation;
+use xcm::{latest::prelude::*, VersionedMultiLocation, WrapVersion};
+use xcm::v2::Response;
+use xcm_mock::{parachain::PALLET_BALANCES_INDEX, *};
+use xcm_simulator::TestExt;
+
+use crate::xcm_mock::parachain::AssetManager;
+
+// `reserved_transfer_asset` contains the following 4 instructions
+//  1. ReserveAssetDeposited(assets.clone()),
+//  2. ClearOrigin,
+//  3. BuyExecution { fees, weight_limit: Limited(0) },
+//  4. DepositAsset { assets: Wild(All), max_assets, beneficiary },
+//  each instruction's weight is 1000, thus, the total weight is 4000
+const RESERVE_TRANSFER_WEIGHT: u64 = 4000;
+
+fn calculate_fee(units_per_seconds: u128, weight: u64) -> u128 {
+	units_per_seconds * (weight as u128) / (WEIGHT_PER_SECOND as u128)
+}
```

Comment:
We should make this a library function, since it is also used by [runtime/primitives/src/xcm.rs](https://github.com/Manta-Network/Manta/pull/382/files?diff=unified&w=0#diff-3366367f9696d8f74ba385cd02fe3c1e4b292086096acb4894da9cc12ce1fa3d).

Reply:
This logic is not in our code. So I would like to get a stand alone code here.